### PR TITLE
tests: Remove Align from Descriptor Buffers

### PR DESCRIPTION
**Pull request by spencer-lunarg**
@artem-lunarg to respond to your comment in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8849/files#r1838581557

### CODE REVIEW

File: tests/unit/descriptor_buffer_positive.cpp [hunk 1]
```diff
@@ -377,8 +372,6 @@ TEST_F(PositiveDescriptorBuffer, MultipleBinding) {
     VkDeviceSize ds_layout_size = 0;
     vk::GetDescriptorSetLayoutSizeEXT(device(), ds_layout.handle(), &ds_layout_size);
 
-    ds_layout_size = Align(ds_layout_size, descriptor_buffer_properties.descriptorBufferOffsetAlignment);
```
[the remark this answers is not in the document]
Comment (artem-lunarg) [reply tests/unit/descriptor_buffer_positive.cpp:380]: The spec says:
> The precise location accessed by a shader for a given descriptor is as follows:
location = bufferAddress + setOffset + descriptorOffset + (arrayElement × descriptorSize)

It looks like queried offset does not have additional alignment requirements or is it about other components of that formula?

**Review by artem-lunarg — APPROVED** [review]
LGTM since I can't spot an issue, but would be nice to get confirmation from @jeremyg-lunarg

File: tests/unit/descriptor_buffer_positive.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply tests/unit/descriptor_buffer_positive.cpp:380]: when this failed CI I was assuming that it meant descriptorSize was not alligned. If the size is supposed to be aligned to descriptorBufferOffsetAlignment then we're good. It is likely for most implementation that the size and offset alignment are equal.

[the remark this answers is not in the document]
Comment (artem-lunarg) [reply tests/unit/descriptor_buffer_positive.cpp:380]: My thinking was that offset is returned by the API itself, so if there is no explicit alignment requirements the implementation takes care about it.

**Review by jeremyg-lunarg — APPROVED** [review]

File: tests/unit/descriptor_buffer_positive.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply tests/unit/descriptor_buffer_positive.cpp:380]: Yes I was being overly paranoid

[the remark this answers is not in the document]
Comment (artem-lunarg) [reply tests/unit/descriptor_buffer_positive.cpp:380]: Right attitude in memory unsafe languages.

**Pull request merged**